src/main/java/cn/nukkit/block/Block.java

```diff
@@ -53,64 +61,132 @@ public static Vector3i fromKey(long key) {
         Block block = BlockRegistry.get().getBlock(identifier, meta).clone();
-        block.x = x;
-        block.y = y;
-        block.z = z;
-        block.level = level;
+        block.position = Vector3i.from(x, y, z);
         block.layer = layer;
+        block.level = level;
         return block;
     }
 
     @Deprecated
     public static Block get(int id) {
-        return get(id, 0, null, 0, 0, 0);
+        return get(id, 0, 0, 0, 0, null);
     }
 
     @Deprecated
     public static Block get(int id, int meta) {
-        return get(id, meta, null, 0, 0, 0);
+        return get(id, meta, 0, 0, 0, null);
     }
 
     @Deprecated
-    public static Block get(int id, int meta, Position pos) {
-        return pos != null ? get(id, meta, pos.level, (int) pos.x, (int) pos.y, (int) pos.z) :
-                get(id, meta, null, 0, 0, 0);
+    public static Block get(int id, int meta, Vector4i pos) {
+        return pos != null ? get(id, meta, pos.getX(), pos.getY(), pos.getZ(), pos.getW(), null) :
+                get(id, meta, 0, 0, 0, null);
     }
 
     @Deprecated
-    public static Block get(int id, int meta, Level level, int x, int y, int z) {
+    public static Block get(int id, int meta, int x, int y, int z, Level level) {
+        return get(id, meta, x, y, z, 0, level);
+    }
+
+    @Deprecated
+    public static Block get(int id, int meta, int x, int y, int z, int layer, Level level) {
         Block block = BlockRegistry.get().getBlock(id, meta).clone();
-        block.x = x;
-        block.y = y;
-        block.z = z;
+        block.position = Vector3i.from(x, y, z);
+        block.layer = layer;
         block.level = level;
         return block;
     }
 
-    public boolean place(Item item, Block block, Block target, BlockFace face, Vector3f clickPos, Player player) {
-        return this.getLevel().setBlock(this, this, true, true);
+    //http://minecraft.gamepedia.com/Breaking
+    private static float breakTime0(float blockHardness, boolean correctTool, boolean canHarvestWithHand,
+                                    Identifier id, int toolType, int toolTier, int efficiencyLoreLevel, int hasteEffectLevel,
+                                    boolean insideOfWaterWithoutAquaAffinity, boolean outOfWaterButNotOnGround) {
+        float baseTime = ((correctTool || canHarvestWithHand) ? 1.5f : 5.0f) * blockHardness;
+        float speed = 1.0f / baseTime;
+        boolean isWoolBlock = id == WOOL, isCobweb = id == WEB;
+        if (correctTool) speed *= toolBreakTimeBonus0(toolType, toolTier, isWoolBlock, isCobweb);
+        speed += speedBonusByEfficiencyLore0(efficiencyLoreLevel);
+        speed *= speedRateByHasteLore0(hasteEffectLevel);
+        if (insideOfWaterWithoutAquaAffinity) speed *= 0.2f;
+        if (outOfWaterButNotOnGround) speed *= 0.2f;
+        return 1.0f / speed;
+    }
+
+    private static float toolBreakTimeBonus0(
```

Comment:
0? For what?

Reply:
This isn't new, just sorted around a bit. Appending 0 to the end of your method is a way to mark it as internal in Java.